### PR TITLE
Shared address component for Provider/Workshop

### PR DESCRIPTION
**Pull request by litvinets**
1. removed old functionality regarding address creation
2. fixed some minor workshop errors

### CODE REVIEW

File: src/app/shell/personal-cabinet/provider/create-address-form/create-address-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  get buildingNumberFormControl(): FormControl {
+    return this.addressFormGroup.get('buildingNumber') as FormControl;
+  }
+
+  ngOnInit(): void {
+    this.initSettlementListener();
+    this.activateEditMode();
+  }
+
+  activateEditMode(): void {
```
Comment (adv1) [inline src/app/shell/personal-cabinet/provider/create-address-form/create-address-form.component.ts:72]: can be private

File: src/app/shell/personal-cabinet/provider/create-address-form/create-address-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      );
+    }
+  }
+
+  /**
+   * This method listen input changes and handle search
+   */
+  private initSettlementListener(): void {
+    merge(
+      this.settlementSearchFormControl.valueChanges,
```
Comment (adv1) [inline src/app/shell/personal-cabinet/provider/create-address-form/create-address-form.component.ts:160]: no need to merge 2 streams, from the same source

File: src/app/shell/personal-cabinet/provider/create-address-form/create-address-form.component.html [hunk 3]
```diff
@@ -0,0 +1,99 @@
+
```
Comment (adv1) [inline src/app/shell/personal-cabinet/provider/create-address-form/create-address-form.component.html:1]: empty row

File: src/app/shell/personal-cabinet/provider/create-address-form/create-address-form.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
```
Comment (adv1) [inline src/app/shell/personal-cabinet/provider/create-address-form/create-address-form.component.spec.ts:2]: empty row

File: src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts [hunk 5]
```diff
@@ -1,95 +1,61 @@
...
-  Validators.minLength(ValidationConstants.INPUT_LENGTH_3),
-  Validators.maxLength(ValidationConstants.INPUT_LENGTH_60)
-];
 
 @Component({
   selector: 'app-create-contacts-form',
   templateUrl: './create-contacts-form.component.html',
-  styleUrls: ['./create-contacts-form.component.scss']
+  styleUrls: ['./create-contacts-form.component.scss'],
 })
```
Comment (adv1) [inline src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts:12]: how about OnPush  strategy for components?

File: src/app/shell/personal-cabinet/provider/create-workshop/create-workshop-address/create-workshop-address.component.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  readonly validationConstants = ValidationConstants;
+
+  @Input() address: Address;
+  @Output() passAddressFormGroup = new EventEmitter();
+
+  addressFormGroup: FormGroup;
+  searchFormGroup: FormGroup;
+
+  constructor(private formBuilder: FormBuilder) {
+    this.addressFormGroup = this.formBuilder.group({
```
Comment (adv1) [inline src/app/shell/personal-cabinet/provider/create-workshop/create-workshop-address/create-workshop-address.component.ts:21]: its can be moved to the OnInit

File: src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts (hunk 5 above)
Comment (litvinets) [reply src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts:13]: will be implemented in the scope of separate task where OnPush will be provided everywhere

File: src/app/shell/personal-cabinet/provider/create-address-form/create-address-form.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { MatAutocompleteModule } from '@angular/material/autocomplete';
+import { MatOptionModule } from '@angular/material/core';
+import { MatInputModule } from '@angular/material/input';
+import { CreateAddressFormComponent } from './create-address-form.component';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { NgxsModule } from '@ngxs/store';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+fdescribe('CreateAddressFormComponent', () => {
```
Comment (adv1) [inline src/app/shell/personal-cabinet/provider/create-address-form/create-address-form.component.spec.ts:12]: fdescribe

File: src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts [hunk 8]
```diff
@@ -101,63 +68,10 @@ export class CreateContactsFormComponent implements OnInit, OnDestroy {
     this.initData();
   }
 
-  /**
-   * This method handle displayed value for mat-autocomplete dropdown
-   * @param codeficator: Codeficator | string
-   */
-  displaySettlementNameFn(codeficator: Codeficator | string): string {
-    return typeof codeficator === 'string' ? codeficator : codeficator?.settlement;
+  click(val: string): string {
```
Comment (adv1) [inline src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts:71]: redurant function

File: src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.html [hunk 9]
```diff
@@ -1,142 +1,34 @@
...
-  <ng-container *ngTemplateOutlet="addressControlsTmlp;
-                                     context: {
-                                        $implicit: {
-                                           addressFormGroup: legalAddressFormGroup,
-                                           streetControl: streetLegalFormControl,
-                                           buildingNumberControl: buildingNumberLegalFormControl
-                                        }
-                                     }"></ng-container>
+<div class="step" fxLayout="column" fxLayoutAlign="center space-between">
+  <h4 class="step-title">Юридична адреса {{ click('Andrey') }}</h4>
```
Comment (adv1) [inline src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.html:3]: redurant function click()

File: src/app/shell/personal-cabinet/provider/provider.module.ts [hunk 10]
```diff
@@ -25,10 +24,12 @@ import { SharedCabinetModule } from '../shared-cabinet/shared-cabinet.module';
 import { RouterModule } from '@angular/router';
 import { FormsModule } from '@angular/forms';
 import { WorkingHoursFormWrapperComponent } from './create-workshop/create-about-form/working-hours-form-wrapper/working-hours-form-wrapper.component';
+import { CreateAddressFormComponent } from './create-address-form/create-address-form.component';
+import { CreateWorkshopAddressComponent } from './create-workshop/create-workshop-address/create-workshop-address.component';
 @NgModule({
```
Comment (adv1) [inline src/app/shell/personal-cabinet/provider/provider.module.ts:29]: it seems that a empty line is missing

**Review by adv1 — APPROVED** [review]

**Pull request merged**